src/moscot/problems/cross_modality/_translation.py

```diff
@@ -0,0 +1,292 @@
+    QuadInitializer_t,
+    ScaleCost_t,
+)
+from moscot.base.problems.compound_problem import B, CompoundProblem, K
+from moscot.base.problems.problem import OTProblem
+from moscot.problems._utils import handle_cost, handle_joint_attr
+from moscot.problems.cross_modality._mixins import CrossModalityTranslationMixin
+from moscot.utils.subset_policy import DummyPolicy, ExternalStarPolicy
+
+__all__ = ["TranslationProblem"]
+
+
+class TranslationProblem(CompoundProblem[K, OTProblem], CrossModalityTranslationMixin[K, OTProblem]):
+    """
+    Class for integrating single cell multiomics data, based on :cite:`demetci-scot:22`.
+
+    Parameters
+    ----------
+    adata_src
+        Instance of :class:`anndata.AnnData` containing the data of the source distribution.
+    adata_tgt
+        Instance of :class:`anndata.AnnData` containing the data of the target distribution.
+    """
+
+    def __init__(self, adata_src: AnnData, adata_tgt: AnnData, **kwargs: Any):
+        super().__init__(adata_src, **kwargs)
+        self._adata_tgt = adata_tgt
+
+    def _create_policy(  # type: ignore[override]
+        self,
+        policy: Literal["external_star"] = "external_star",
+        key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Union[DummyPolicy, ExternalStarPolicy[K]]:
+        """Private class to create DummyPolicy if no batches are present in the spatial anndata."""
+        del policy
+        if key is None:
+            return DummyPolicy(self.adata, **kwargs)
+        return ExternalStarPolicy(self.adata, key=key, **kwargs)
+
+    def _create_problem(
+        self,
+        src: K,
+        tgt: K,
+        src_mask: ArrayLike,
+        tgt_mask: ArrayLike,
+        **kwargs: Any,
+    ) -> OTProblem:
+        return self._base_problem_type(
+            adata=self.adata_src,
+            adata_tgt=self.adata_tgt,
+            src_obs_mask=src_mask,
+            tgt_obs_mask=None,
+            src_key=src,
+            tgt_key=tgt,
+            **kwargs,
+        )
+
+    def prepare(
+        self,
+        src_attr: Union[str, Mapping[str, Any]],
+        tgt_attr: Union[str, Mapping[str, Any]],
+        joint_attr: Optional[Union[str, Mapping[str, Any]]] = None,
+        batch_key: Optional[str] = None,
+        cost: OttCostFnMap_t = "sq_euclidean",
+        cost_kwargs: CostKwargs_t = types.MappingProxyType({}),
+        a: Optional[str] = None,
+        b: Optional[str] = None,
+        **kwargs: Any,
+    ) -> "TranslationProblem[K]":
+        """
+        Prepare the :class:`moscot.problems.cross_modality.TranslationProblem`.
+
+        This method prepares the data to be passed to the optimal transport solver.
+
+        Parameters
+        ----------
+        src_attr
+            - If `str`, it must refer to a key in :attr:`anndata.AnnData.obsm`.
+            - If `dict`, the dictionary stores `attr` (attribute of :class:`anndata.AnnData`) and `key`
+            (key of :class:`anndata.AnnData` ``['{attr}']``).
```

Comment:
This needs to be correctly indented, it causes docs linter to fail.

Reply:
I copied it as it is from `/src/moscot/_docs/_docs.py`. Wonder what's the difference here to that file.

Reply:
see comment above